scripts/calibrateArmDac.sh

```diff
@@ -34,71 +34,79 @@ then
     return
 fi
 
-echo -e "ChamberName\tscandate\tCFG_THR_ARM_DAC" 2>&1 | tee listOfScanDates_calibrateArmDac_${DETECTOR}.txt
+echo -e "ChamberName\tscandate\tCFG_THR_ARM_DAC" 2>&1 | tee ${DATA_PATH}/${DETECTOR}/armDacCal/${scandate}/listOfScanDates_calibrateArmDac_${DETECTOR}.txt
+
+scandate=$(date +%Y.%m.%d.%H.%M)
+mkdir -p ${DATA_PATH}/${DETECTOR}/armDacCal/${scandate}
+ln -sf ${DATA_PATH}/${DETECTOR}/armDacCal/${scandate} ${DATA_PATH}/${DETECTOR}/armDacCal/current
 
 runNum=0
 for armDacVal in $(echo $LIST_ARM_DAC | sed "s/,/ /g")
 do
-    if [ -e scurveLog_ArmDac_${armDacVal}.txt ]
-    then
-        rm scurveLog_ArmDac_${armDacVal}.txt
-    fi
-
     echo "=============Run $runNum============="
     echo "Issuing a link reset before CFG_THR_ARM_DAC=${armDacVal}"
-    echo "gem_reg.py -n ${CARDNAME} -e write 'GEM_AMC.GEM_SYSTEM.CTRL.LINK_RESET 1' 2>&1 | tee scurveLog_ArmDac_${armDacVal}.txt"
-    gem_reg.py -n ${CARDNAME} -e write "GEM_AMC.GEM_SYSTEM.CTRL.LINK_RESET 1" 2>&1 | tee scurveLog_ArmDac_${armDacVal}.txt
+    echo "gem_reg.py -n ${CARDNAME} -e write 'GEM_AMC.GEM_SYSTEM.CTRL.LINK_RESET 1' 2>&1 | tee ${DATA_PATH}/${DETECTOR}/armDacCal/${scandate}/scurveLog_ArmDac_${armDacVal}.txt"
+    gem_reg.py -n ${CARDNAME} -e write "GEM_AMC.GEM_SYSTEM.CTRL.LINK_RESET 1" 2>&1 | tee ${DATA_PATH}/${DETECTOR}/armDacCal/${scandate}/scurveLog_ArmDac_${armDacVal}.txt
     sleep 1
 
     echo "Configuring Detector for CFG_THR_ARM_DAC=${armDacVal}"
-    echo "confChamber.py -c ${CARDNAME} -g ${LINK} --vt1=${armDacVal} --vfatmask=${MASK} --zeroChan --run 2>&1 | tee -a scurveLog_ArmDac_${armDacVal}.txt"
-    confChamber.py -c ${CARDNAME} -g ${LINK} --vt1=${armDacVal} --vfatmask=${MASK} --zeroChan --run 2>&1 | tee -a scurveLog_ArmDac_${armDacVal}.txt
+    echo "confChamber.py -c ${CARDNAME} -g ${LINK} --vt1=${armDacVal} --vfatmask=${MASK} --zeroChan --run 2>&1 | tee -a ${DATA_PATH}/${DETECTOR}/armDacCal/${scandate}/scurveLog_ArmDac_${armDacVal}.txt"
+    confChamber.py -c ${CARDNAME} -g ${LINK} --vt1=${armDacVal} --vfatmask=${MASK} --zeroChan --run 2>&1 | tee -a ${DATA_PATH}/${DETECTOR}/armDacCal/${scandate}/scurveLog_ArmDac_${armDacVal}.txt
     sleep 1
+
+    first_unmasked_vfat=0
+    mask_as_array=`echo "obase=2; ibase=16; \`echo $MASK | awk -F"0x" '{print $2}' | awk '{print toupper($1)}'\`;" | bc | grep -o .`
+    i=0; for vfat in $mask_as_array; do echo $vfat; if [ $vfat == 0 ]; then first_unmasked_vfat=$i; break; fi; i=$(($i+1)) done;
```

Comment:
Running this in my developer area I think there's a syntax error in this line; although I'm not sure where.

Typically it's good to have one statement per line that can help narrow down the problem due to debugging.